LPS-105151 General simplification

csierra opened the pull request.

Commit:
LPS-105151 Set OpenId Configuration Provider by Company

Commit:
LPS-105151 Store and filter OpenId Connect Providers by Company

Commit:
LPS-105151 Fix invocations adding the filter by company

Commit:
LPS-105151 CompanyThreadLocal will not be set because deleted() is invoked asynchronously by config admin

Commit:
LPS-105151 SF

Commit:
LPS-105151 Use object caching for frequently invoked method to avoid unnecessary GC

Commit:
LPS-105151 Insufficient concurrency protection and _companyIdPidMapping is doubly protected because only modified/iterated upon in synchronized blocks

Commit:
LPS-105151 SF Fix generics

Commit:
LPS-105151 Fix NoSuchElementException

It looks like the contract is to return null, so we need to make that explicit in the optional

Commit:
LPS-105151 This need to be ConcurrentHashMap?

Commit:
LPS-105151 SF simplify?

Commit:
LPS-105151 Can we use the concurrentMap to block only when accessing the same keys?

Commit:
LPS-105151 General simplification

I think it is better to store the property map to the PID so we have the configuration available both when adding and removing. This way we can make add and remove symmetric and we can use them when creating the configuration for the first time or when we are updating it.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

Commit:
LPS-105151 Take into account system company

Commit:
LPS-105151 SF rename

Commit:
LPS-105151 Replace and rebuild the complete company configuration

This should bring more gracious recovery from configuration inconsistencies and less synchronization on readings, since the structures and copied on writes

csierra commented:
please see https://github.com/stian-sigvartsen/liferay-portal/pull/248
